GS: Improve autoflush rect checks

refractionpcsx2 opened the pull request:
### Description of Changes
Improves the rect calculation of the current draw and compares it to the last texture position from the previous draw to check for page crossings. 

### Rationale behind Changes
Reduces change of false positives and thus lower autoflush performance. 

### Suggested Testing Steps
Test games that require Autoflush, make sure they still work good.


Also unbreaks Sonic Unleashed when Autoflush is enabled in hardware mode.

Performance metrics compared to Master using GS Dumps at Native x3 with Autoflush enabled unless otherwise stated:

Black goes from 224fps to 400fps
GoW 2 goes from 150fps to 300fps
Jak 2 goes from 60fps to 71fps. At Native this translates to 144fps to 160fps
Scarface goes from 200fps to 468fps
Sniper Elite goes from 107fps to 287fps

AhayriSG commented:
SW - Auto Flush: On
![gs_20220402135003_UEFA Champions League 2006-2007 _NTSC-U__SLUS-21581](https://user-images.githubusercontent.com/59465544/161380059-11b0026d-a46f-4929-98f8-714a3859689a.png)
SW - Auto Flush: Off
![gs_20220402135019_UEFA Champions League 2006-2007 _NTSC-U__SLUS-21581](https://user-images.githubusercontent.com/59465544/161380063-5c4b6c2a-f74c-4f25-8f22-adf77930205e.png)
Master (SW - Auto Flush: On)
https://wiki.pcsx2.net/File:Gs_20220331205217_UEFA_Champions_League_2006-2007_NTSC-U_SLUS-21581.png

refractionpcsx2 commented:
Thanks :) I think I messed up something with the vertex kick functions lol

refractionpcsx2 commented:
Sorry about that, I reverted the experimental change I did last night which was causing problems, once I fixed it (but kept it in place) the was no advantage to it anymore, so I just got rid.
